### PR TITLE
Separate Analytic results

### PR DESCRIPTION
**Pull request by mahaalbashir**
These changes are related to issue #37 which aims at Separate analytic results out from json files.

### CODE REVIEW

**Comment by jim-smith** [comment]
@mahaalbashir 
1: Looking through the notebook, the formative the regression reports are presented in there is different to how they appear in the first thejson, then in turn ther csv files. 
Could we fix those please.

2: You've got three cases covered by try/except blocks for when you check whether the outputs directory
1.  exists
2. doesn't;t exist but can be created
3. Doesn't;t exist and can't be created

In the last case, should we output a warning to that effect and saying "outputs wil be written to current directory instead" then rest the path accordingly

3. It would be better to use the logger to send messages as warnings to the user (and get stored in a log file) than to just use print() statements which can easily get lost.

File: acro/utils.py [hunk 1]
```diff
@@ -64,15 +68,35 @@ def finalise_json(filename: str, results: dict) -> None:
...
         Outputs to write.
     """
-    # convert dataframes to json
+
     outputs: dict = copy.deepcopy(results)
-    for _, output in outputs.items():
+
+    absolute_path = os.path.dirname(os.path.dirname(__file__))
+    relative_path = OUTPUT_DIRECTORY
+    path = os.path.join(absolute_path, relative_path).replace("\\", "/")
```
Comment (rpreen) [inline acro/utils.py:76]: Replacing the slashes (also used in the tests) to make it OS independent feels wrong to me --- wouldn't os.path.normpath or similar be safer for future changes?

File: outputs/output_0.csv [hunk 2]
```diff
@@ -0,0 +1,7 @@
+year,G,N,R,R/G
```
Comment (rpreen) [inline outputs/output_0.csv:1]: I think we should remove the generated outputs from this PR and add the outputs folder to the `.gitignore` so we all don't end up accidentally committing them when we run our local tests.

File: acro/utils.py [hunk 3]
```diff
@@ -64,15 +68,35 @@ def finalise_json(filename: str, results: dict) -> None:
...
+
+    # check if the outputs directory was already created
+    try:
+        os.makedirs(path)
+        print(f"Directory {OUTPUT_DIRECTORY} created successfully")
```
Comment (rpreen) [inline acro/utils.py:81]: Please change the print functions to `logger.info()`. The logger functions should use lazy evaluation so also change the fstring to something like `logger.info("Directory %s created successfully", OUTPUT_DIRECTORY)`

**Review by rpreen — CHANGES_REQUESTED** [review]
Mostly looks good, I've just made some minor suggestions.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,11 @@
 Changes:
 *    .
 
+## Version 0.0.5 (March 09, 2023)
+
+Changes:
```
[the remark this answers is not in the document]
Comment (mahaalbashir) [reply CHANGELOG.md:10]: @rpreen Do you mean by that deleting lines 8,9 and 10?

[the remark this answers is not in the document]
Comment (rpreen) [reply CHANGELOG.md:10]: 6--10 so it's just underneath the existing "Changes" so we have all the newest changes listed there until we create a new release

**Review by rpreen — APPROVED** [review]
Looks good. Thanks for making the changes.

**Pull request merged**